Introduce Quarkus for the Web documentation (#39953)
quarkusio/quarkus: docs/src/main/asciidoc/http-reference.adoc

When Undertow is present, RESTEasy functions as a Servlet filter.
In its absence, RESTEasy operates directly on Vert.x without involving Servlets.

> == Serving static resources

Reviewer: Having a section about static resources definitely makes sense to have in here but this section starts rather abruptly and start referring to compression and other config which is not introduced before later.

Maybe move this down a bit ?

Author: This doc was already there, I just removed the capitalisation and added a link to the web doc

Reviewer: yes you are right - just its starting to "grow" big. not a blocker for now.

> If you are looking to use Quarkus for a web application, look at the xref:web.adoc[Quarkus for the Web] guide.

Reviewer: ```suggestion
If you are looking to use Quarkus for a full web application, look at the xref:web.adoc[Quarkus for the Web] guide.
```

== Serving Static Resources

> If you are looking to use Quarkus for your Web Application, you might want to check the xref:web.adoc[Quarkus for the Web] guide.

Author: https://en.wikipedia.org/wiki/Web_application this is pretty clear to me

> If you are looking to use Quarkus for your Web Application, you might want to check the xref:web.adoc[Quarkus for the Web] guide.

Reviewer: ```suggestion
If you are looking to use Quarkus for your Web Application, look at the xref:web.adoc[Quarkus for the Web] guide.
```

=== From the application jar

To serve static resources from the application jar, you must place them in the `META-INF/resources` directory of your application. This location
was chosen as it is the standard location for resources in `jar` files as defined by the Servlet spec. Even though
Quarkus can be used without Servlet, following this convention allows existing code that places its resources in this
location to function correctly.

[[from-mvnpm]]
=== From MVNPM

If you are using https://mvnpm.org/[mvnpm.org], like the following JQuery one:

[source,xml,role="primary asciidoc-tabs-target-sync-cli asciidoc-tabs-target-sync-maven"]
.pom.xml
----
<dependency>
    <groupId>org.mvnpm</groupId>
    <artifactId>jquery</artifactId>

>     <version>3.7.1</version>

Reviewer 2: Maybe lit ?

>     <version>3.7.1</version>

Author: I just kept the existing example, but I don't mind changing it

>     <version>3.7.1</version>

Author: I switched to bootstrap as lit is better with an import-map

</dependency>
----

[source,gradle,role="secondary asciidoc-tabs-target-sync-gradle"]
.build.gradle
----
implementation("org.mvnpm:jquery:3.7.1")
----

You can use it in your html like this:
[source,html]
----

> <script src="/_static/jquery/3.7.1/dist/jquery.min.js"></script>

Author: @phillip-kruger  is working on this, we will update it as soon as we have the web-dependency-locator for mvnpm

> <script src="/_static/jquery/3.7.1/dist/jquery.min.js"></script>

Author: https://github.com/quarkusio/quarkus/pull/39963

Quarkus can be used without Servlet, following this convention allows existing code that places its resources in this
location to function correctly.

[[from-mvnpm]]
=== From mvnpm

> If you are using https://mvnpm.org/[mvnpm], as for the following JQuery dependency:

Reviewer: is all this available from vert-http extension ? 

the doc actually dont state which extension you need to get these features (old bug but now that there is more added wanted to make sure it is clear)
